src/shareclient.cpp

```diff
@@ -146,6 +147,96 @@ std::string ShareClient::Commit() {
     }
 }
 
+/*
+ * Add function to support streams (curl_mimepart)
+ * Reference: https://curl.se/libcurl/c/curl_mime_data_cb.html
+ *
+ */
+
+DDB_DLL int ShareClient::StartUploadSession(int chunks, size_t size) {
+    if (token.empty())
+        throw InvalidArgsException("Missing token, call Init first");
+
+    if (chunks < 1) throw InvalidArgsException("Chunks cannot be less than 1");
+    if (size <= 0) throw InvalidArgsException("Invalid size");
+
+    // Commit
+    int retryNum = 0;
+    while (true) {
+        try {
+            auto res =
+                net::POST(this->registry->getUrl("/share/upload/" +
+                                                 this->token + "/session"))
+                    .multiPartFormData({"chunks", std::to_string(chunks)},
+                                       {"size", std::to_string(size)})
+                    .authToken(this->registry->getAuthToken())
+                    .send();
+
+            if (res.status() != 200) this->registry->handleError(res);
+
+            auto j = res.getJSON();
+            if (!j.contains("sessionId")) this->registry->handleError(res);
+
+            return j["sessionId"].get<int>();
+
+        } catch (const NetException& e) {
+            if (++retryNum >= MAX_RETRIES) throw e;
+            LOGD << e.what() << ", retrying start upload session (attempt "
+                 << retryNum << ")";
+            utils::sleep(1000 * retryNum);
+        }
+    }
+}
+
+DDB_DLL void ShareClient::UploadToSession(int sessionId, int index,
+                                          std::istream input) {
+    return;
+}
+
+DDB_DLL void ShareClient::CloseUploadSession(int sessionId, std::string& path) {
+    if (token.empty())
+        throw InvalidArgsException("Missing token, call Init first");
+
+    if (path.empty()) throw InvalidArgsException("Missing path");
+
+    // Commit
+    int retryNum = 0;
+    while (true) {
+        try {
+            auto res =
+                net::POST(this->registry->getUrl(
+                              "/share/upload/" + this->token + "session/" +
+                              std::to_string(sessionId) + "/close"))
+                    .multiPartFormData({"path", path})
+                    .authToken(this->registry->getAuthToken())
+                    .send();
+
+            if (res.status() != 200) this->registry->handleError(res);
+
+            json j = res.getJSON();
+
+            if (!j.contains("hash")) this->registry->handleError(res);
+
+            // Find a way to include this check
```

Comment:
This can probably be checked when the last chunk has been sent and registry has merged all the chunks.

(This one is kind of important to avoid corruption).

Reply:
Added in 2bcb9da